velox/functions/sparksql/specialforms/FromJson.cpp

```diff
@@ -267,6 +272,52 @@ struct ExtractJsonTypeImpl {
     }
   };
 
+  static simdjson::error_code castJsonToDate(
+      Input value,
+      exec::GenericWriter& writer) {
+    SIMDJSON_ASSIGN_OR_RAISE(auto type, value.type());
+    if (type != simdjson::ondemand::json_type::string) {
+      return simdjson::INCORRECT_TYPE;
+    }
+    std::string_view s;
+    SIMDJSON_ASSIGN_OR_RAISE(s, value.get_string());
+    int32_t day = 0;
+    // If the value has fewer than four digits, it is interpreted as the number
+    // of days since January 1, 1970.
+    if (s.size() < 4) {
+      auto result = folly::tryTo<int32_t>(s);
+      if (!result.hasError()) {
+        day = result.value();
+      } else {
+        return simdjson::INCORRECT_TYPE;
+      }
+    } else {
+      auto days =
+          util::fromDateString(StringView(s), util::ParseMode::kSparkCast);
+      if (!days.hasError()) {
+        day = days.value();
```

Comment:
nit: the names of `day` and `days` might be confusing as their difference is not the singular or plural form.